Source/JavaScriptCore/wasm/WasmConstExprGenerator.cpp

```diff
@@ -43,10 +47,70 @@ class ConstExprGenerator {
     using ErrorType = String;
     using PartialResult = Expected<void, ErrorType>;
     using UnexpectedResult = Unexpected<ErrorType>;
-    using ExpressionType = uint64_t;
-    using ResultList = Vector<ExpressionType, 8>;
     using CallType = CallLinkInfo::CallType;
 
+    // Represents values that a constant expression may evaluate to.
+    // If a constant expression allocates an object, it should be put in a Strong handle.
+    struct ConstExprValue {
```

Comment:
I believed I've fixed the GC issues now, by changing `ExpressionType` to use a new struct `ConstExprValue`. It can either hold bits for a non-object or a `Strong<JSObject>` handle for keeping an allocated object alive.

I wanted to write a specific independent test that triggered the GC failure, but it seems the best way to trigger the failure is to run the existing tests in the continuous-gc mode (these should now pass too).